R/helpers.R

```diff
@@ -113,25 +113,30 @@ extract_num_iterations <- function(x) {
   length(evals)
 }
 
-# Given the result of a CV search, get the max number of iterations from the
+# Given the result of a CV search, get the number of iterations from the
 # result set with the best performing hyperparameters
-select_max_iterations <- function(tune_results, metric) {
-  dplyr::bind_cols(
-    tune_results %>%
-      dplyr::select(id, .metrics, .extracts) %>%
-      tidyr::unnest(cols = .metrics) %>%
-      dplyr::filter(.metric == params$cv$best_metric) %>%
-      dplyr::select(-.extracts),
-    tune_results %>%
-      tidyr::unnest(cols = .extracts) %>%
-      tidyr::unnest(cols = .extracts) %>%
-      dplyr::select(.extracts)
-  ) %>%
-    dplyr::inner_join(
-      tune::select_best(tune_results, metric = metric),
-      by = ".config"
+select_iterations <- function(tune_results, metric, type = "mean") {
+  stopifnot(type %in% c("mean", "median", "max"))
+  func <- switch(type,
+    mean = mean,
+    median = median,
+    max = max
+  )
+
+  tune_results %>%
+    dplyr::select(id, .metrics, .extracts) %>%
+    tidyr::unnest(cols = .metrics) %>%
+    dplyr::filter(.metric == params$cv$best_metric) %>%
+    dplyr::select(-.extracts) %>%
+    dplyr::left_join(
+      tune_results %>%
+        tidyr::unnest(cols = .extracts) %>%
+        tidyr::unnest(cols = .extracts) %>%
```

Comment:
**[Question, non-blocking]** The transformation in question is a bit above my head so I'm not super concerned about it, but is there a reason we need to unnest this twice in a row?

Reply:
Yes! It's a doubly nested column containing a tibble that contains a list. So gotta call it twice.